packages/docusaurus-theme-classic/src/theme/CodeBlock/index.tsx

```diff
@@ -90,11 +91,16 @@ export default function CodeBlock({
     languageProp ?? parseLanguage(blockClassName) ?? prism.defaultLanguage;
   const {highlightLines, code} = parseLines(content, metastring, language);
 
-  const handleCopyCode = () => {
+  const handleCopyCode = (e: MouseEvent<HTMLButtonElement>) => {
     copy(code);
-    setShowCopied(true);
+    setIsCopied(true);
 
-    setTimeout(() => setShowCopied(false), 2000);
+    const button = e.currentTarget;
+
+    setTimeout(() => {
```

Comment:
Aren't there cases where we should clean this timeout? (like multiple copy clicks or whatever?)

In general it's not a very good practice to not handle timeout cleanups as it can lead to weird race conditions

Reply:
Not sure, I actually reverted these edits that were previously made to the handler as result of another refactoring.